### DISCUSSION TITLE
如何增加自定义的组件？

### DISCUSSION THREAD

**Discussion by williampang2021** [alibaba/designable#56] [Q&A]
如何增加自定义的组件？  比如业务自己的个性化组件

**Comment by Pioneer-Linzi** [comment]
同问哈，研究了半天也没看出来

**Comment by Pioneer-Linzi** [comment]
const DesignableField = createDesignableField({
  registryName: 'DesignableField',
  components: {  },
})


在 components 里注册

**Reply by robin536180** [reply]
createDesignableField方法从哪里引 @Pioneer-Linzi

**Reply by robin536180** [reply]
是在index里面注册吗
createDesignableField方法中components是我们定义的自定义组件吗

**Comment by xjchenhao** [comment]
增加自定义组件：https://www.yuque.com/xjchenhao/development/dhv90n
@Pioneer-Linzi @williampang2021 @robin536180

**Comment by YqxLzx** [comment]
那怎么自定义setting的组件呢

**Reply by scil** [reply]
看示例就行，直接复制就可。我觉得麻烦的是，在组件中建立子组件，不过终于搞定了
